tools/themis/src/main.rs

```diff
@@ -0,0 +1,74 @@
+use cargo_metadata::camino::Utf8PathBuf;
+
+mod rules;
+mod style;
+#[macro_use]
+mod utils;
+
+#[derive(Debug)]
+pub struct Package {
+    parsed: cargo_metadata::Package,
+    raw: toml::Value,
+}
+
+#[derive(Debug)]
+pub struct Workspace {
+    root: Utf8PathBuf,
+    members: Vec<Package>,
+}
+
+#[derive(Debug)]
+pub struct PackageOutcome<'a> {
+    pkg: &'a Package,
+    value: Option<String>,
+}
+
+#[derive(Debug)]
+pub struct Expected {
+    value: String,
+    reason: Option<String>,
+}
+
+#[derive(Debug)]
+pub enum Error<'a> {
+    OutcomeError { msg: String, expected: Option<Expected>, outliers: Vec<PackageOutcome<'a>> },
+    RuntimeError(anyhow::Error),
+}
+
+impl<'a, E: Into<anyhow::Error>> From<E> for Error<'a> {
+    fn from(err: E) -> Self {
+        Error::RuntimeError(err.into())
+    }
+}
+
+fn main() -> anyhow::Result<()> {
+    let workspace = utils::parse_workspace()?;
+
+    if workspace.members.is_empty() {
+        return Ok(());
+    }
```

Comment:
Hm, why do we need this condition here, it seems like if we remove it than everything will just work? If anything, seems better to assert that the list of memebers is not empty here, to sanity check that we, in fact, do check something.

Reply:
Yeah, that makes sense.